File: funcscript/parser/syntax/FuncScriptParser.SkipSpace.cs
namespace FuncScript.Core
{
    public partial class FuncScriptParser
    {
        record SkipSpaceResult(int NextIndex);

        static SkipSpaceResult SkipSpace(ParseContext context, int index)
        {
            int i = index;
            var expression = context.Expression;
            while (index < expression.Length)
            {
                if (IsCharWhiteSpace(expression[index]))
                {
                    index++;
                }
                else
                {
                    var commentBlockResult = GetCommentBlock(context, index);
                    if (commentBlockResult.NextIndex == index)
                        break;
                    index = commentBlockResult.NextIndex;
                }
            }

            return new SkipSpaceResult(index);
        }
    }

    public record SkipSpaceResult(int NextIndex);
}

File: ts/core/src/parser/syntax/FuncScriptParser.SkipSpace.ts
import { ParseContext, ParseNode, ParseNodeType, SyntaxErrorData } from "../FuncScriptParser.Main";
import { isCharWhiteSpace } from "../FuncScriptParser.Utils";
import { GetCommentBlock } from "./FuncScriptParser.GetCommentBlock";

export function SkipSpace(context: ParseContext, index: number): { NextIndex: number } {
    let i = index;
    const expression = context.Expression;
    
    while (index < expression.length) {
        if (isCharWhiteSpace(expression[index])) {
            index++;
        } else {
            const commentBlockResult = GetCommentBlock(context, index);
            if (commentBlockResult.NextIndex === index) {
                break;
            }
            index = commentBlockResult.NextIndex;
        }
    }

    return { NextIndex: index };
}